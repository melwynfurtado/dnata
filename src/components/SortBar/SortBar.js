import React from 'react'
import PropTypes from 'prop-types'
import Link from './Link'

const SortBar = ({ handleOnSort, sortBy }) => {
  return (
    <nav className="sortbar navbar navbar-expand-lg navbar-light">
      <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#sortBy">
        Sort by <span className="fa fa-sort"></span>
      </button>
      <div className="collapse navbar-collapse" id="sortBy">
        <div className="navbar-nav">
          <Link clickHandle={handleOnSort} sortBy={sortBy} field="Distance" label="Distance" />
          <Link clickHandle={handleOnSort} sortBy={sortBy} field="Stars" label="Stars" />
          <Link clickHandle={handleOnSort} sortBy={sortBy} field="MinCost" label="Minimum Cost" />
          <Link clickHandle={handleOnSort} sortBy={sortBy} field="UserRating" label="User ratings" />
        </div>
      </div>
    </nav>    
  )
}

SortBar.propTypes = {
  handleOnSort: PropTypes.func,
  sortBy: PropTypes.shape({
    order: PropTypes.string,
  }),
}

export default SortBar
