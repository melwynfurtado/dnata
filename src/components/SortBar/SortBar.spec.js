import React from 'react'
import renderer from 'react-test-renderer'
import SortBar from './SortBar'

describe('SortBar Component', () => {
  it('renders correctly', () => {
    const Component = <SortBar handleOnSort={()=>{}} sortBy={{}} />

    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })
})