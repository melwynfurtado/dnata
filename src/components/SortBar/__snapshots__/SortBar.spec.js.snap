// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar Component renders correctly 1`] = `
<nav
  className="sortbar navbar navbar-expand-lg navbar-light"
>
  <button
    className="navbar-toggler"
    data-target="#sortBy"
    data-toggle="collapse"
    type="button"
  >
    Sort by 
    <span
      className="fa fa-sort"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="sortBy"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link "
        field="Distance"
        href="#"
        onClick={[Function]}
        order="undefined"
      >
        Distance
      </a>
      <a
        className="nav-item nav-link "
        field="Stars"
        href="#"
        onClick={[Function]}
        order="undefined"
      >
        Stars
      </a>
      <a
        className="nav-item nav-link "
        field="MinCost"
        href="#"
        onClick={[Function]}
        order="undefined"
      >
        Minimum Cost
      </a>
      <a
        className="nav-item nav-link "
        field="UserRating"
        href="#"
        onClick={[Function]}
        order="undefined"
      >
        User ratings
      </a>
    </div>
  </div>
</nav>
`;
